### PR TITLE
[php-symfony] Symfony6 support

### LINKED ISSUE #11322: [REQ][PHP][SYMFONY] Upgrade php-symfony code generation to work witH Symfony6
### Is your feature request related to a problem? Please describe.

Code generate for symfony is not fully compatible with Symfony6 and/or php8.1. Some little hacks/workarounds are mandatory to use generated code in sf6.

## Describe the solution you'd like

I would apply here the udpdates I am using to fix the problem. I am already using this generator with some fixes. I would like to create a pull request here and stop to use my workaround.

## Describe alternatives you've considered

I've considered the alternative to just open an issue and stay until someone solve this problem.

## Additional context

Not actually. I would like to open the pull request and start to contribute to the php part of this project.

### PR DESCRIPTION
**Pull request by sensorario**
Update php-symfony's generator template to work with Symfony6 and php8. fix #11322 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Hi @jebentier, @dkarlovi, @mandrean, @jfastnacht, @ackintosh, @ybelenko, @renepardon! I've made this PR because I needed to generate php code from a swagger file. I am using Symfony8 and php8. Hope this Pull request is good enough or a starting point to improve the symfony support of this project.

### CODE REVIEW

**Comment by wing328** [comment]
@sensorario thanks for the PR. Is it correct to say that this is a breaking change? If that's case, should this PR target 6.0.x branch (breaking changes without fallback)?

**Comment by sensorario** [comment]
@wing328 yes. May I rebase into 6.0.x, ...

**Comment by sensorario** [comment]
@wing328 I think something went wrong. I checked out branch 6.0.x. I moved all my changes here. Recreated symfony6-support branch from here and then followed the steps 

```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```

But now I see all samples generated.

**Comment by sensorario** [comment]
> You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example ./bin/generate-samples.sh bin/configs/java*.

For this PR is goot just php-symfony or may I add all other updated examples?

**Comment by ybelenko** [comment]
Cool, finally someone ready to share modern Symfony build. Let me review, I want to test it locally before merge. Thanks @sensorario

**Comment by wing328** [comment]
> maybe keep the old version with creating new generator like php-symfony-6?

Sounds good to me.

**Comment by sensorario** [comment]
I guess something is not clear for me. I checked out 6.0.x to, started new branch from here. [...] Then, ... the command `./bin/generate-samples.sh` fails. May I open new issue? Should the sequence 

```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```

must works in 6.0.x branch?

**Comment by wing328** [comment]
@sensorario I've changed the PR target branch to "master" (which is the upcoming v6.0.0 release). Can you please resolve the merge conflicts when you've time?

File: modules/openapi-generator/src/main/resources/php-symfony/composer.mustache [hunk 1]
```diff
@@ -19,22 +19,24 @@
...
-        "php": "^7.1.3",
+        "php": ">=8.0.2",
         "ext-curl": "*",
         "ext-json": "*",
         "ext-mbstring": "*",
-        "symfony/validator": "*",
-        "jms/serializer-bundle": "^2.0",
-        "symfony/framework-bundle": "^4.4.8"
+        "symfony/validator": "6.0.*",
+        "jms/serializer-bundle": "6.0.*",
```
Comment (BenjaminHae) [inline modules/openapi-generator/src/main/resources/php-symfony/composer.mustache:27]: Latest available version is 4.0.2
https://packagist.org/packages/jms/serializer-bundle

**Comment by BenjaminHae** [comment]
Additionally, I think some work needs to be done for compatibility with the new JMS/Serializer version.
As a first step the changes introduced by this PR https://github.com/OpenAPITools/openapi-generator/pull/10763 can be taken.
However this targets version 3.5. for version 4.0 some more work is needed.

**Comment by ybelenko** [comment]
> Additionally, I think some work needs to be done for compatibility with the new JMS/Serializer version.
> As a first step the changes introduced by this PR #10763 can be taken.
> However this targets version 3.5. for version 4.0 some more work is needed.

Well, I thought this PR contains significant changes(when I noticed updated files list 😅), but it's just bump of the Symfony and PHP number. I'm afraid that we have to start Symfony 6 generator from scratch. I'm not even sure that current codegen builds and works.

@sensorario did you compare generator output with basic Symfony 6 skeleton(fresh installation)? I think they completely different.

**Comment by BenjaminHae** [comment]
@ybelenko I did the required changes to make it really work with symfony6 in #11810

**Comment by ybelenko** [comment]
@sensorario thanks for the effort. I'm closing PR without merge in favour of #11810 let's focus in the one thread.

**Pull request closed without merging**